pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -118,15 +118,19 @@ func (r *DiffListResult) Render(opt DiffRenderOptions) string {
 		index++
 		b.WriteString(fmt.Sprintf("# %d. %s\n\n", index, key.ReadableString()))
 
-		if opt.UseDiffCommand {
+		// Use our diff check in the following cases:
+		// - not explicit set useDiffCommand option.
+		// - requires masking secret or configmap value.
+		if !opt.UseDiffCommand || opt.MaskSecret || opt.MaskConfigMap {
```

Comment:
I think we have to use `diff` command even if `opt.MaskSecret` is true and the resource is not a Secret.

Reply:
Ah, I get your point, will update the condition

Reply:
So basically we need a flag configured at lines 112 and 114.

Reply:
Btw, could you add an issue about adding more tests around this part?

Reply:
PTAL https://github.com/pipe-cd/pipe/pull/2423/commits/c76d206ba37ac29efd13ae6280c8b39bac3bfc8b